include/openenclave/internal/switchless.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef _OE_SWITCHLESS_H
+#define _OE_SWITCHLESS_H
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/bits/types.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/thread.h>
+
+typedef struct _host_worker_thread_context
+{
+    volatile oe_call_host_function_args_t* call_arg;
+    bool is_stopping;
+    oe_enclave_t* enclave;
+} host_worker_thread_context_t;
+
+typedef struct _oe_switchless_call_manager
+{
+    host_worker_thread_context_t* host_worker_contexts;
+    oe_thread_t* host_worker_threads;
+    size_t num_host_workers;
```

Comment:
nit: I'd recommend "host_worker_count", but this file isn't public so not as important as if it were public